src/instance.cc

```diff
@@ -33,10 +33,14 @@ InstanceReader::InstanceReader(const Configuration& config)
     : config_(config), environment_(config) {}
 
 /*static*/
-MonitoredResource InstanceReader::InstanceResource(const Environment& environment) {
+MonitoredResource InstanceReader::InstanceResource(const Environment& environment)
+    throw(std::out_of_range) {
   const std::string resource_type = environment.InstanceResourceType();
   const std::string instance_id = environment.InstanceId();
   const std::string zone = environment.InstanceZone();
+  if (instance_id.empty() || zone.empty()) {
+    throw std::out_of_range("No instance information");
```

Comment:
Nit: is `out_of_range` the right type of exception for this? We seem to be using `QueryException` for `GetLegacyResource`, should these be consistent?

Reply:
`std::out_of_range` a standard exception is thrown by a container when a value is not found. I propose to treat `InstanceReader` as a container and `InstanceResource` as a lookup.
`QueryException` is internal to `KubernetesReader`.

Reply:
https://github.com/Stackdriver/metadata-agent/blob/2e1b70d1526f748fb60a95a677cc3ccec82b1dcf/src/kubernetes.cc#L388 seems to have the same exact type of validation, but uses a different type of exception, so call this a bit of a rant, but it led me to be confused when digesting the code.